client/asset/eth/multirpc.go

```diff
@@ -502,6 +508,44 @@ func (m *multiRPCClient) connect(ctx context.Context) (err error) {
 	return nil
 }
 
+// unusedNonce returns true and saves the nonce for the next call when a nonce
+// has not been received recently.
+func (m *multiRPCClient) unusedNonce(nonce uint64) bool {
```

Comment:
The `registerNonce` name certainly fits better with the method's functionality.  Less fond of error returns when a flag serves the same purpose.

Reply:
unusedNonce -> registerNonce

I don't feel an error is great because none of the outcomes are an error imo, unless the provider continues to give us a nonce we are sure we used. From testnet testing, we expect the same nonce to come back once sometimes from remote providers.